libs/utils/analysis/thermal_analysis.py

```diff
@@ -0,0 +1,243 @@
+        res = df['cpus'].unique().tolist()
+        return [mask_to_list(mask) for mask in res]
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+
+    def _dfg_thermal_zone_temperature(self, ids=None):
+        """
+        Get the temperature readings of one or more thermal zone(s).
+
+        :param ids: The thermal zones to consider
+        :type ids: list(int)
+        """
+        df = self._dfg_trace_event('thermal_temperature')
+        df = df[['id', 'thermal_zone', 'temp']]
+
+        if ids is not None:
+            df = df[df.id.isin(ids)]
+
+        return df
+
+    def _dfg_cpufreq_cooling_state(self, cpus=None):
+        """
+        Get the cpufreq cooling device states
+
+        :param cpus: The cpus to consider
+        :type cpus: list(int)
+        """
+        df = self._dfg_trace_event('thermal_power_cpu_limit')
+        df = df[['cpus', 'freq', 'cdev_state']]
+
+        if cpus is not None:
+            # Find masks that match the requested CPUs
+            # This can include other CPUs
+            masks = self._matchingMasks(cpus)
+            df = df[df.cpus.isin(masks)]
+
+        return df
+
+    def _dfg_devfreq_cooling_state(self, devices=None):
+        """
+        Get the temperature readings of one or more thermal zone(s).
+
+        :param devices: The devfreq devices to consider
+        :type device: list(str)
+        """
+        df = self._dfg_trace_event('thermal_power_devfreq_limit')
+        df = df[['type', 'freq', 'cdev_state']]
+
+        if devices is not None:
+            df = df[df.type.isin(devices)]
+
+        return df
+
+
+###############################################################################
+# Plotting Methods
+###############################################################################
+
+    def plotTemperature(self, thermal_zones=None, ax=None):
+        """
+        Plot temperature of thermal zones (all by default)
+
+        Requires the following trace event:
+            - thermal_temperature
+
+        :param thermal_zones: ID(s) of the zones to be plotted.
+            All the zones are plotted by default.
+            IDs can be found in syfs:  /sys/class/thermal/thermal_zone<ID>
+        :type thermal_zones: list(int)
+        """
+        if not self._trace.hasEvents('thermal_temperature'):
+            self._log.warning('Event [{}] not found, plot DISABLED!'.format(event))
+            return
+
+        plot_df = self._dfg_thermal_zone_temperature(thermal_zones)
+
+        def stringify_tz(id):
+            return plot_df[plot_df.id == id]['thermal_zone'].unique()[0]
+
+        plot_generic(plot_df, 'id', names = thermal_zones,
+                     fields = ['temp'], prettify_name = stringify_tz,
+                     xlim = (self._trace.x_min, self._trace.x_max),
+                     drawstyle = 'steps-post', ax = ax
```

Comment:
Can we make the nomenclature here match trappy's ILinePlot API? Also, no spaces around `=` inside parameter lists please.

Reply:
> Can we make the nomenclature here match trappy's ILinePlot API?

Do you mean matching the order of the parameters, or going the `**kwargs` route ?

Reply:
I mean the actual names of the parameters - see the penultimate paragraph of [this comment](https://github.com/ARM-software/trappy/pull/269#discussion_r140769563)

Reply:
Ah gotcha, skipped that one when I closed the PR